torch/utils/data/datapipes/iter/combining.py

```diff
@@ -40,6 +40,26 @@ def __len__(self) -> int:
         return len(self)
 
 
+# This is fake class to show API, going to be replaced by the copy from torchdata
+# TODO(VitalyFedyunin): Replace with valid version, documentation and tests
+class IterateBuffer(IterDataPipe):
+    def __init__(self, buffer):
+        self.buffer = buffer
+
+    def __iter__(self):
+        for i in self.buffer:
+            yield i
+
+
+@functional_datapipe('fork')
+class ForkIterDataPipe(IterDataPipe):
+
+    def __new__(cls, datapipe, instances):
+        result = []
+        buffer = [i for i in datapipe]
```

Comment:
We should be able to use `list(datapipe)` now after fix the error from `__len__`.